src/main/java/frc/robot/superstructure/Superstructure.java

```diff
@@ -127,55 +185,95 @@ public Command speaker() {
+			new SequentialCommandGroup(
+				new ParallelCommandGroup(
+					elevatorRollerStateHandler.setState(ElevatorRollerState.STOP),
+//					swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.AMP)),
+					elevatorStatesHandler.setState(ElevatorStates.AMP)
+				).until(this::isReadyToAmp),
+				new ParallelCommandGroup(
+//					swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.AMP)),
+					elevatorRollerStateHandler.setState(ElevatorRollerState.AMP),
+					elevatorStatesHandler.setState(ElevatorStates.AMP)
+				),//.until(() -> !isNoteInElevatorRoller()),
+				new ParallelCommandGroup(
+					elevatorRollerStateHandler.setState(ElevatorRollerState.STOP),
+					elevatorStatesHandler.setState(ElevatorStates.IDLE)
+				)
+			),
 			flywheelStateHandler.setState(FlywheelState.DEFAULT),
-			elevatorRollerStateHandler.setState(ElevatorRollerState.AMP),
-			funnelStateHandler.setState(FunnelState.AMP),
+			intakeStatesHandler.setState(IntakeStates.STOP),
 			pivotStateHandler.setState(PivotState.UP),
-			elevatorStatesHandler.setState(ElevatorStates.AMP)
+			funnelStateHandler.setState(FunnelState.STOP)
 		);
 	}
 
 	public Command transferShooterElevator() {
 		return new ParallelCommandGroup(
-			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
+			setCurrentStateValue(RobotState.TRANSFER_SHOOTER_ELEVATOR),
+			new SequentialCommandGroup(
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.INTAKE),
+					funnelStateHandler.setState(FunnelState.SHOOTER_TO_ELEVATOR),
+					elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_TO_ELEVATOR)
+				),//.until(this::isNoteInElevatorRoller),
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.STOP),
+					funnelStateHandler.setState(FunnelState.STOP),
+					elevatorRollerStateHandler.setState(ElevatorRollerState.STOP)
+				)
+			),
+//			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
 			flywheelStateHandler.setState(FlywheelState.DEFAULT),
-			elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_TO_ELEVATOR),
-			funnelStateHandler.setState(FunnelState.SHOOTER_TO_ELEVATOR),
-			intakeStatesHandler.setState(IntakeStates.SHOOTER_TO_ELEVATOR),
-			pivotStateHandler.setState(PivotState.UP),
-			elevatorStatesHandler.setState(ElevatorStates.IDLE)
+			elevatorStatesHandler.setState(ElevatorStates.IDLE),
+			pivotStateHandler.setState(PivotState.UP)
 		);
 	}
 
 	public Command transferElevatorShooter() {
 		return new ParallelCommandGroup(
-			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
+			setCurrentStateValue(RobotState.TRANSFER_ELEVATOR_SHOOTER),
+			new SequentialCommandGroup(
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.NOTE_TO_SHOOTER),
+					funnelStateHandler.setState(FunnelState.NOTE_TO_SHOOTER)
+				),//.until(this::isNoteInShooter),
```

Comment:
in other cases you've lest time to make sure the note isn't in the stopped subsystem anymore. idk if we have to do that here too, but pls check that out. 

Also, why this is commented out